packages/lpa-questionnaire-web-app/src/controllers/upload-plans.js

```diff
@@ -8,3 +9,80 @@ exports.getUploadPlans = (req, res) => {
     uploadedFiles: [],
   });
 };
+
+exports.postUploadPlans = async (req, res) => {
+  const documents = req.body?.files?.documents || [];
+
+  const { delete: deleteFile = '', errors = {}, submit = '' } = req.body;
+
+  // Chance for delete to be triggered due to non-JS solution. delete will be set to value of filename if button clicked
+  if (deleteFile) {
+    // TODO: add handling for deletion from DB and doc store as part of AS-1538
+    const file = req.session.uploadedFiles?.find((upload) => upload.name === deleteFile);
+
+    if (file) {
+      req.session.uploadedFiles = req.session.uploadedFiles.filter(
+        (upload) => upload.name !== deleteFile
+      );
+    }
+  } else if (Array.isArray(documents) && documents.length) {
+    // Chance for files to be attached due to non-JS solution, these need passing into session uploaded files (with appropriate errors);
+    req.session.uploadedFiles = [
+      ...(req.session.uploadedFiles || []),
+      ...documents.map((doc, index) => ({
+        ...doc,
+        error: errors[`files.documents[${index}]`]?.msg,
+      })),
+    ];
+  }
+
+  const errorMessage = errors.documents && errors.documents.msg;
+
+  // due to the way express validator handles errors, and the fact that file errors persist after a post we need to build a custom summary here
+  const constructedErrorSummary = [
+    ...(errorMessage
+      ? [
+          {
+            href: '#documents',
+            text: errorMessage,
+          },
+        ]
+      : []),
+    ...(req.session.uploadedFiles
+      ? req.session.uploadedFiles.reduce((errorsOutput, file) => {
+          if (file.error)
+            errorsOutput.push({
+              href: `#${file.name}`,
+              text: file.error,
+            });
+          return errorsOutput;
+        }, [])
+      : []),
+  ];
```

Comment:
wondering if this could be moved off to a separate function, as this controller is > 80 lines at this point, and seemingly going to get bigger due to line 63

Reply:
I'm considering modularising the whole function as the next task, given it is generic and will be reused about a dozen times. Happy for this to be part of a different task or this PR?

Reply:
Sounds good - surprised we haven't had to do the JS variant yet on upload. Wonder if that's been forgotten actually. I'll raise it.